client/src/main/java/io/pravega/client/stream/impl/Pinger.java

```diff
@@ -38,7 +31,7 @@
  */
 @Slf4j
 public class Pinger implements AutoCloseable {
-    private static final double PING_INTERVAL_FACTOR = 0.5; //ping interval = factor * txn lease time.
+    private static final double PING_INTERVAL_FACTOR = 0.25; //ping interval = factor * txn lease time.
```

Comment:
For low txn lease times, this could make pings lot more aggressive, which will take up additional compute resources on both controller and client. 
We should either have a floor on the ping interval so that we know that we are not hammering the server with very aggressive pings. 
The intent here is to make sure that "at least" one ping per txn-lease-time period is sent so that transaction's lease gets renewed. So maybe 0.5 factor suffices, and we could just add retry block on failed pings (and only for retryable failures).

Reply:
Ideally, pings should be acked and retransmitted in the case an ack doesn't come back on time. It sounds fine to start a new ping at 0.5 of the interval, but then we should retransmit more aggressively in the absence of an ack.

Another way to look at this is that we are using gRPC, which builds on TCP. TCP has retransmission, so we shouldn't have to retransmit any ACK, unless there is an error bubbling up from gRPC.

Reply:
Done